Libplanet/Blockchain/BlockChain.cs

```diff
@@ -542,19 +537,20 @@ public void StageTransactions(IImmutableSet<Transaction<T>> transactions)
         }
 
         /// <summary>
-        /// Removes <paramref name="transactions"/> from the pending list.
+        /// Removes <paramref name="transaction"/> from the pending list.
         /// </summary>
-        /// <param name="transactions"><see cref="Transaction{T}"/>s
+        /// <param name="transaction">A <see cref="Transaction{T}"/>
         /// to remove from the pending list.</param>
-        /// <seealso cref="StageTransactions"/>
-        public void UnstageTransactions(ISet<Transaction<T>> transactions)
+        /// <seealso cref="StageTransaction"/>
+        public void UnstageTransaction(Transaction<T> transaction)
         {
+            // FIXME it's global chain lock so using it in this method can cause degrading
+            // parallelism of `BlockChain<T>`. we should re-organize locks in `BlockChain<T>`
             _rwlock.EnterWriteLock();
 
             try
             {
-                Store.UnstageTransactionIds(
-                    transactions.Select(tx => tx.Id).ToImmutableHashSet());
+                Store.UnstageTransactionIds(new[] { transaction.Id }.ToImmutableHashSet());
```

Comment:
Using [`ImmutableHashSet.Create<T>(T)` method][1] might be slightly more efficient for a single element.

```suggestion
                Store.UnstageTransactionIds(ImmutableHashSet.Create<TxId>(transaction.Id));
```

[1]: https://docs.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutablehashset.create?view=netcore-3.1&viewFallbackFrom=netstandard-2.1#System_Collections_Immutable_ImmutableHashSet_Create__1___0_